src/Components/shared/ProtectedRoute/ProtectedRoute.tsx

```diff
@@ -0,0 +1,10 @@
+import { Navigate } from "react-router-dom";
+
+export default function ProtectedRoute({
+  children,
+}: {
+  children: React.ReactNode;
+}) {
+  if (localStorage.getItem("token") ) return children;
+  else return <Navigate to="/login" />;
```

Comment:
It is better to use a useEffect to handle the navigation instead of using the Navigate component. This component is intended for use in component classes where it is not possible to use hooks.
![image](https://github.com/user-attachments/assets/8ff7838c-8aec-4b32-a778-591673b22d04)